pkgdb/src/buildenv/realise.cc

```diff
@@ -60,36 +56,164 @@ namespace flox::buildenv {
+#endif
+
+#ifndef FLOX_PROCPS_PKG
+#  error "FLOX_PROCPS_PKG must be set to the path of the nix procps package"
 #endif
 
 /* -------------------------------------------------------------------------- */
 
+// Top-level activate script, always invoked with nix bash.
+const char * const ACTIVATE_SCRIPT = R"_(
+# Flox environment activation script.
+[ "${_FLOX_PKGDB_VERBOSITY:-0}" -eq 0 ] || set -x
+
+# Capture starting environment.
+_start_env="$($_coreutils/bin/mktemp --suffix=.start-env)"
+_end_env="$($_coreutils/bin/mktemp --suffix=.end-env)"
+export | $_coreutils/bin/sort > "$_start_env"
+
+function cleanup() {
+  $_coreutils/bin/rm -f "$_start_env" "$_end_env"
+}
+
+# Set FLOX_ENV as the path by which all flox scripts can make reference to
+# the environment to which they belong. Use this to define the path to the
+# activation scripts directory.
+FLOX_ENV="$( $_coreutils/bin/dirname -- "${BASH_SOURCE[0]}" )"
+export FLOX_ENV
+
+# Process the flox environment customizations, which includes (amongst
+# other things) prepending this environment's bin directory to the PATH.
+if [ -d "$FLOX_ENV/etc/profile.d" ]; then
+  declare -a _prof_scripts;
+  _prof_scripts=( $(
+    cd "$FLOX_ENV/etc/profile.d";
+    shopt -s nullglob;
+    echo *.sh;
+  ) );
+  for p in "${_prof_scripts[@]}"; do . "$FLOX_ENV/etc/profile.d/$p"; done
+  unset _prof_scripts;
+fi
+
+# Set static environment variables from the manifest.
+if [ -f "$FLOX_ENV/activate.d/envrc" ]; then
+  source "$FLOX_ENV/activate.d/envrc"
+fi
+
+# Source the hook-on-activate script if it exists.
+if [ -e "$FLOX_ENV/activate.d/hook-on-activate" ]; then
+  # Nothing good can come from output printed to stdout in the
+  # user-provided hook scripts because these can get interpreted
+  # as configuration statements by the "in-place" activation
+  # mode. So, we'll redirect stdout to stderr.
+  source "$FLOX_ENV/activate.d/hook-on-activate" 1>&2
+elif [ -e "$FLOX_ENV/activate.d/hook-script" ]; then
+  # [hook.script] is deprecated - print warning and source it.
+  echo "WARNING: [hook.script] is deprecated, use [hook.on-activate] instead." >&2
+  source "$FLOX_ENV/activate.d/hook-script" 1>&2
+fi
+
+# From this point on the activation process depends on the mode:
+
+# 1. "command" mode: simply exec the provided command and args
+if [ $# -gt 0 ]; then
+  cleanup
+  # The colon is a no-op command that returns true so don't
+  # attempt to execute it as a command.
+  if [ "$1" = ":" ]; then
+    exit 0
+  else
+    exec "$@"
+  fi
+fi
+
+# The remaining modes require that $FLOX_SHELL be set by the rust CLI.
+[ -n "$FLOX_SHELL" ] || {
+  echo "FLOX_SHELL not set .. defaulting to ${SHELL}" >&2
+  FLOX_SHELL="${SHELL}"
+}
+
+# 2. "interactive" mode: invoke the user's shell with args that:
+#   a. defeat the shell's normal startup scripts
+#   b. source the relevant activation script
+if [ -t 1 -o -n "$FLOX_FORCE_INTERACTIVE" ]; then
```

Comment:
```suggestion
if [ -t 1 -o -n "$_FLOX_FORCE_INTERACTIVE" ]; then
```
Per Yannik's suggestion.